### PR TITLE
Fix transformer example and update doc

### LINKED ISSUE #1659: [attention example]An error occurred while calling o44.connectInputs.
```
Run attention example would occur a problem as below
Traceback (most recent call last):
  File "/home/aqtjin/analytics-zoo/pyzoo/zoo/examples/attention/transformer.py", line 71, in <module>
    O_seq = GlobalAveragePooling1D()(O_seq)
  File "/home/aqtjin/ENTER/envs/tf_env/lib/python3.7/site-packages/zoo/pipeline/api/keras/base.py", line 43, in __call__
    to_list(x)))
  File "/home/aqtjin/ENTER/envs/tf_env/lib/python3.7/site-packages/bigdl/util/common.py", line 589, in callBigDlFunc
    raise e
  File "/home/aqtjin/ENTER/envs/tf_env/lib/python3.7/site-packages/bigdl/util/common.py", line 585, in callBigDlFunc
    result = callJavaFunc(api, *args)
  File "/home/aqtjin/ENTER/envs/tf_env/lib/python3.7/site-packages/bigdl/util/common.py", line 634, in callJavaFunc
    result = func(*args)
  File "/home/aqtjin/ENTER/envs/tf_env/lib/python3.7/site-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/home/aqtjin/ENTER/envs/tf_env/lib/python3.7/site-packages/py4j/protocol.py", line 328, in get_return_value
    format(target_id, ".", name), value)
py4j.protocol.Py4JJavaError: An error occurred while calling o44.connectInputs.
: java.lang.RuntimeException: Invalid operation
	at com.intel.analytics.bigdl.utils.Shape$class.toSingle(Shape.scala:25)
	at com.intel.analytics.bigdl.utils.MultiShape.toSingle(Shape.scala:81)
	at com.intel.analytics.zoo.pipeline.api.keras.layers.GlobalAveragePooling1D.doBuild(GlobalAveragePooling1D.scala:45)
	at com.intel.analytics.bigdl.nn.keras.KerasLayer.build(KerasLayer.scala:225)
	at com.intel.analytics.bigdl.nn.keras.KerasLayer.inputs(KerasLayer.scala:244)
	at com.intel.analytics.zoo.pipeline.api.keras.python.PythonZooKeras.connectInputs(PythonZooKeras.scala:1164)
	at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by hkvision**
- Fix #1659 #1660 
- Expose documentation on website
- Fix some doc format and typos.

### CODE REVIEW

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.211:18888/job/ZOO-PR-Validation/2677/

File: docs/docs/KerasStyleAPIGuide/Layers/self-attention.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-entirely. Refer https://s3-us-west-2.amazonaws.com/openai-assets/research-covers/language-unsupervised/language_understanding_paper.pdf
+entirely. Refer to this [paper](https://s3-us-west-2.amazonaws.com/openai-assets/research-covers/language-unsupervised/language_understanding_paper.pdf) for more details.
 
 Input is a Table which consists of 2 tensors.
+
 1. Token id tensor: shape (batch, seqLen) with the word token indices in the vocabulary
 2. Position id tensor: shape (batch, seqLen) with positions in the sentence.
 
-Output is a Tensor which output the states of Transformer layer
+Output are the states of Transformer layer and the pooled result of the first token.
```
Comment (dding3) [inline docs/docs/KerasStyleAPIGuide/Layers/self-attention.md]: ***and the pool output which process the last layer hidden-state of the first token of the sequence and necessary for segment-level tasks.

File: docs/docs/KerasStyleAPIGuide/Layers/self-attention.md [hunk 2]
```diff
@@ -232,14 +235,15 @@ Output is
...
+
 1. Token id tensor: shape (batch, seqLen) with the word token indices in the vocabulary
 2. Token type id tensor: shape (batch, seqLen) with the token types in (0, 1).
    0 means `sentence A` and 1 means a `sentence B` (see BERT paper for more details).
 3. Position id tensor: shape (batch, seqLen) with positions in the sentence.
 4. Attention_mask tensor: shape (batch, seqLen) with indices in (0, 1).
   It's a mask to be used if the input sequence length is smaller than seqLen in the current batch.
 
-Output is an Activity which output the states of BERT layer
+Output are the states of BERT layer and the pooled result of the first token.
```
Comment (dding3) [inline docs/docs/KerasStyleAPIGuide/Layers/self-attention.md]: Output are *** and the pool output which process the last layer hidden-state of the first token of the sequence and necessary for segment-level tasks.

**Comment by dding3** [comment]
Thank you very much for the fix. It would be great if scala doc can also be fixed :)
https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/TransformerLayer.scala#L38

**Comment by hkvision** [comment]
@dding3 All updated.

**Comment by hkvision** [comment]
jenkins passed http://10.239.47.211:18888/job/ZOO-PR-Validation/2681/
Thank you for your review.

**Pull request merged**